test/time.test.ts

```diff
@@ -114,5 +127,49 @@ describe("time helpers", function () {
         )
       })
     })
+
+    describe("waitForTransaction function", function () {
+      let ethers: typeof ethersT & HardhatEthersHelpers
+      let signer: HardhatEthersSigner
+      let transactionResponse: ethersT.TransactionResponse
+
+      beforeEach(async function () {
+        ethers = this.hre.ethers
+        signer = (await ethers.getSigners())[0]
+
+        const tx = {
+          to: signer.address, // Sending the transaction to the signer's own address
+          value: ethers.parseEther("0.01"), // Small amount of Ether, can also be 0
+        }
+        transactionResponse = await signer.sendTransaction(tx)
+      })
+
+      it("returns immediately on hardhat", async function () {
+        const waitResult = await Promise.race([
+          timeHelpers.waitForTransaction(transactionResponse.hash, 5),
+          timeout(300),
+        ])
+        expect(waitResult).to.be.true
+      })
+
+      it("does not return immediately on non-hardhat networks", async function () {
+        this.hre.network.name = "not-hardhat"
```

Comment:
Out of curiosity how changing network name make this transaction take longer? 😄

Reply:
the first big of logic in `waitForTransaction` is:

```typescript
  if (hre.network.name === "hardhat") {
    return true
  }
```
  
The default for tests is "hardhat", so changing it to anything else causes that branch to run and for the waiting to stop

Reply:
I'll add a comment!

Reply:
b71b4e98ffed6db1f35c914a917f0922fc409fab